### PR TITLE
[KAN-15] About Page Second Section

### PR DESCRIPTION
**Pull request by eunicey**
PR does not include responsive states (will be done in a separate PR). 
There's certain styling bugs that I noted in my comments.
Screenshot 2024-07-01 at 12 53 25 PM

### CODE REVIEW

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [inline src/components/About/ProfileCard.module.css]: I can't figure out why the id selectors are not working. If I add the background to the .button class, it's just fine.

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [inline src/components/About/ProfileCard.module.css]: Similar issue as above. The icons are suppose to change in color but I can't seem to get it to work.

File: src/constants/directors.js
Comment (eunicey) [inline src/constants/directors.js]: I added filler text for 4 out of the 6 personas since I could not find their content in the figma file.

File: src/components/utility/Typography/Typography.module.css
Comment (eunicey) [inline src/components/utility/Typography/Typography.module.css]: these are supposed to be 18px (at root font size 16).

File: src/components/utility/Typography/Typography.module.css
Comment (eunicey) [inline src/components/utility/Typography/Typography.module.css]: Not trying to be nit picky here but it makes it easier to figure out the source of spacing discrepancies.

File: src/components/utility/Typography/Typography.module.css
Comment (joanwolk) [reply src/components/utility/Typography/Typography.module.css]: Please revert this. The default styles here are declared for the _mobile_ display size. The desktop font size is in the media query on line 55 (and is using two decimal places after discussion between Poulami and me on the Typography PRs).

File: src/components/utility/Typography/Typography.module.css
Comment (joanwolk) [reply src/components/utility/Typography/Typography.module.css]: Styles for `.body2 `and `.body1` are the same at mobile sizes (where this is declared) but they differ because `.body1` will be a larger font-size on the desktop view. You want `body1` if you're looking for the font that will be 18px on desktop.

**Tip:** If you're changing things to be sure you're seeing the right styles applied, change them in a BIG way. Make a font 3x larger, or add a bright red background, or something that will be _very_ different from the expected style. It will help you see it faster, and it will help ensure that a debugging change won't make its way into production.

Please revert this change.

File: src/components/About/ProfileCard.jsx
Comment (joanwolk) [inline src/components/About/ProfileCard.jsx]: Do not use `id` here. This component will be on the page multiple times, and an `id` should appear on the page exactly once.

File: src/components/About/ProfileCard.jsx
Comment (joanwolk) [inline src/components/About/ProfileCard.jsx]: Same here, don't use `id`

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [reply src/components/About/ProfileCard.module.css]: Take a look at the Button.jsx and Button.module.css files to see how both the base class (`.button`, via `styles.button`) and a secondary class (`.buttonPrimary` for example, via `styles.buttonPrimary`) are being applied, and how the hover states are styled.

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [reply src/components/About/ProfileCard.module.css]: Two steps to make these work:

1. Use the SVG directly in the HTML instead of importing it as an img
2. Change the selectors to `.twitter:hover path` and `.linkedin:hover path`

You can also combine the two rules like this:
```
.linkedin:hover path,
.twitter:hover path {
  fill: #F67E7E;
}
```

File: src/components/pages/About/About.jsx
Comment (joanwolk) [inline src/components/pages/About/About.jsx]: If you want to make adding the other components easy for whoever comes after you, wrap this in a [Fragment](https://react.dev/reference/react/Fragment):

```
return (
  <>
    <Directors />
  </>
);
```

This is because the return statement has to be one element (that can contain any number of other elements). The use of React Fragments lets us return one thing without adding unnecessary `div`s to the structure.

**Review by joanwolk — CHANGES_REQUESTED** [review]

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [reply src/components/About/ProfileCard.module.css]: And as noted in the JSX file, these should be classes, not ids.

File: src/components/utility/Typography/Typography.module.css
Comment (eunicey) [reply src/components/utility/Typography/Typography.module.css]: Changes reverted, sorry about that!

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [reply src/components/About/ProfileCard.module.css]: Thank you for the guidance @joanwolk! Let me know if I fixed the code in the way that you expected.

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [reply src/components/About/ProfileCard.module.css]: Thank you so much!

File: src/components/About/ProfileCard.jsx
Comment (eunicey) [reply src/components/About/ProfileCard.jsx]: I did not know that! Thanks!

File: src/components/pages/About/About.jsx
Comment (eunicey) [reply src/components/pages/About/About.jsx]: Done!

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [inline src/components/About/ProfileCard.module.css]: Lines 57 and 58 are perfectly cancelling each other out—try disabling them both in the browser to see it—so you can delete those two lines.

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [inline src/components/About/ProfileCard.module.css:53]: Currently, the 500 weight of the font is not being imported, so the display doesn't actually reflect this.

The Figma design does use medium italic here (500 weight) despite it not being mentioned in the typography specs. To update the font import, change the import URL in line 1 of `App.css` to `https://fonts.googleapis.com/css2?family=Livvic:ital,wght@0,500;0,600;0,700;1,500;1,600;1,700&display=swap` (adding the regular and italic for weight 500)

**Review by joanwolk — COMMENTED** [review]
Overall, the changes look great! There's a problem now with the color of the show/hide buttons, but that seems to be a problem with the use of `style` in the `Button` component so @poulami-saha should take a look at how to make that more flexible, or else decide to add the appropriate color patterns to the `elType` options.

There are a couple of small tweaks for the existing styles, but you're definitely ready to work on the responsiveness. 👍 

Screenshot 2024-07-06 at 1 07 19 PM

File: src/components/About/Clients.module.css
Comment (eunicey) [inline src/components/About/Clients.module.css]: With mobile/table widths part of the image goes in the background and the directors section goes in the foreground.

File: src/components/About/Directors.module.css
Comment (eunicey) [inline src/components/About/Directors.module.css:76]: I was just curious as to why we use 960 instead of 1440? The desktop design seems to be made for 1440 so parts of the left and right profile cards are hidden at 960.

File: src/components/About/directorsData.js
Comment (eunicey) [inline src/components/About/directorsData.js:1]: I moved this file to be consistent with the SuccessStories component and how the data is in the same folder.

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [reply src/components/About/ProfileCard.module.css]: I removed them.

File: src/components/About/Directors.module.css
Comment (joanwolk) [reply src/components/About/Directors.module.css:76]: There are a variety of approaches to where breakpoints are set, but the important thing is to remember that most views in a category will not be at exactly the size shown on the design. In the case of desktops, 1440px is both quite a bit narrower than most desktop (and laptop) displays, and quite a bit wider than most tablet displays. Remember that 1080px is where HD starts, so that's already quite wide! 

960px is larger than all the common tablets, so it's a reasonable (and common) point to choose for the desktop breakpoint. You'll note that the Figma designs are at 1440px wide (desktop), 768px wide (tablet), and 375px wide (mobile). We have to be aware of sizes in between all of those—and occasionally larger than 1440px, though not in this design.

There are multiple approaches to how to fix the display of the cards at 960px width. One would be to make the cards more responsive; another would be to make the layout of the cards more fluid so it automatically wraps to two cards per line if there's overflow.

File: src/components/About/Directors.module.css
Comment (joanwolk) [inline src/components/About/Directors.module.css]: To address the card size issue on small desktops and ensure there's some gutter space (empty space on the left and right edges of the page), do this AND the change in ProfileCard.module.css for the desktop card width:
```
  max-width: 94vw;
  margin: 5.5rem auto;
```

File: src/components/About/ProfileCard.module.css
Comment (joanwolk) [inline src/components/About/ProfileCard.module.css]: To address the card size issue on small desktops, do this AND the change in Directors.module.css for the article width:
```
    width: calc((96vw / 3) - 1.875rem); /* 1/3 container width, minus gap width */
    max-width: 21.875rem;
```

**Review by joanwolk — CHANGES_REQUESTED** [review]
Here's my take on how to fix the layout for small desktops 🙂

File: src/components/About/Clients.module.css
Comment (joanwolk) [reply src/components/About/Clients.module.css]: This is a reasonable change. However, there have been changes merged to main recently that are going to conflict with this because we changed from using img tags in the HTML for the decorative graphics to using `::before` and `::after` pseudo-elements with the images in the CSS. See the [new Clients.module.css implementation](https://github.com/poulami-saha/Global-Code-Queens-MyTeam-Website/blob/69ae621c3edc5a1a9430d2302f57661cc4224c4a/src/components/About/Clients.module.css#L19-L25).

You can revert this line, and ideally also update the Directors component to use the same pattern.

**Comment by joanwolk** [comment]
Nice work on fixing the button colors! Looks like you fixed the application of a class in the ProfileCard to solve it? (/cc @poulami-saha)

Screenshot 2024-07-09 at 10 22 13 AM

File: src/components/About/Directors.module.css
Comment (eunicey) [reply src/components/About/Directors.module.css:76]: Thanks for the explanation! I've changed the cards to be displayed as flex instead of grid so that the cards wrap. The row/column gaps between the cards differ proportionately for the design sizes so I'm not sure what they should be for intermittent sizes. Another weird thing is that the card width in the design is smallest at 768px so I can't consistently scale the cards as the width increases. However, I've ensured that the number of cards per row consistently increases with width.

**Comment by eunicey** [comment]
> Nice work on fixing the button colors! Looks like you fixed the application of a class in the ProfileCard to solve it? (/cc @poulami-saha)
> 
> Screenshot 2024-07-09 at 10 22 13 AM

I couldn't find a fix until the last commit (maybe). I noticed that when I first render the card component, the background color seems to render but once I cause an error in the rendering, it stops working. In the latest changes, I made the path more specific for the custom styling and hopefully it holds. @poulami-saha - let me know if there is a better way to do this.

File: src/components/About/ProfileCard.module.css
Comment (eunicey) [reply src/components/About/ProfileCard.module.css]: Hopefully my card changes from displaying grid to flex doesn't affect change. From what I can tell, it doesn't.

File: src/App.css
Comment (poulami-saha) [inline src/App.css]:
```suggestion
  --very-pale-cyan: #C4FFFE;
```

File: src/components/About/Directors.module.css
Comment (poulami-saha) [inline src/components/About/Directors.module.css:3]:
```suggestion
  margin: 0;
  box-sizing: border-box;
```

File: src/components/About/Clients.module.css
Comment (eunicey) [inline src/components/About/Clients.module.css:13]: I added this to prevent the image from bleeding into the Director's component at mobile size.
Screenshot 2024-07-10 at 11 49 11 AM

File: src/components/About/Directors.module.css
Comment (poulami-saha) [inline src/components/About/Directors.module.css:4]: Wont the above suggestion works the same way as this?

Comment (eunicey) [reply src/components/About/Directors.module.css:4]: Thank you, I'll make that change!

File: src/components/pages/About/About.jsx
Comment (poulami-saha) [inline src/components/pages/About/About.jsx]: I think finally this can be removed as we no longer need a styling for the entire page

File: src/components/About/ProfileCard.jsx
Comment (poulami-saha) [inline src/components/About/ProfileCard.jsx:26]: const avatarContainer = () => (
    <>
      
      <Typography elType="h3" className={styles.name}>
        {name}
      </Typography>
      <Typography elType="body2" className={styles.title}>
        {title}
      </Typography>
    </>
  );

  const quoteContainer = () => (
    <>
      <Typography elType="h3" className={styles.name}>
        {name}
      </Typography>
      <Typography elType="body2" className={styles.quote}>
        {`"${quote}"`}
      </Typography>
      <div>
        <TwitterIcon className={styles.twitterIcon} />
        <LinkedinIcon className={styles.linkedinIcon} />
      </div>
    </>
  );

Comment (poulami-saha) [inline src/components/About/ProfileCard.jsx:26]: `const avatarContainer = () => (
    <>
      
      <Typography elType="h3" className={styles.name}>
        {name}
      </Typography>
      <Typography elType="body2" className={styles.title}>
        {title}
      </Typography>
    </>
  );

  const quoteContainer = () => (
    <>
      <Typography elType="h3" className={styles.name}>
        {name}
      </Typography>
      <Typography elType="body2" className={styles.quote}>
        {`"${quote}"`}
      </Typography>
      <div>
        <TwitterIcon className={styles.twitterIcon} />
        <LinkedinIcon className={styles.linkedinIcon} />
      </div>
    </>
  );`

**Comment by eunicey** [comment]
@poulami-saha, @joanwolk - I made changes that you requested. Please let me know what you think. I still can't figure out why my custom button styling doesn't always work. For example, if I cause a compiling error and then fix it, my .mainButton styling is ignored. 
Screenshot 2024-07-14 at 4 35 27 PM
Screenshot 2024-07-14 at 4 36 04 PM

**Comment by joanwolk** [comment]
I've fixed the functionality of the button colors, but I want to refactor the code for it.

For one thing, having the button colors as style code directly on the element makes it very hard to customize  with CSS where it would be appropriate to do so (like your button colors here). It could be done by adding an `!important` declaration to the end of your background color rules, but that would be both aggressive and brittle (because other people will start modifying things with !important, which is bad news).

For another thing, it's putting style concerns in the HTML and JS, which is not where they belong.

**Review by joanwolk — COMMENTED** [review]
I think this could be merged before I do the refactor of the button styles, since we're short on time, but I'd like @poulami-saha to sign off on this PR too before merging.

**Review by joanwolk — APPROVED** [review]

**Comment by joanwolk** [comment]
@poulami-saha told me it's fine to make the refactor a separate PR, so let's merge this! 👍

**Pull request merged**
